### ISSUE TITLE
std.array.split cannot deduce function from argument types when compiled with gdc + linux

### ISSUE THREAD

**Issue by truedat101** [NCrashed/serial-port#9]
```
../../../../../../../../home/dkords/.dub/packages/serial-port-1.1.0/source/serial/device.d:922: error: template std.array.split cannot deduce function from argument types !("msecs")(Duration), candidates are:
/mnt/lotsostuff/dev/tools/x86_64-gdcproject-linux-gnu/include/d/4.9.0/std/array.d:1358: note: std.array.split(S)(S s) if (isSomeString!S)
/mnt/lotsostuff/dev/tools/x86_64-gdcproject-linux-gnu/include/d/4.9.0/std/array.d:1438: note: std.array.split(R, E)(R r, E delim) if (isForwardRange!R && is(typeof(ElementType!R.init == E.init)))
/mnt/lotsostuff/dev/tools/x86_64-gdcproject-linux-gnu/include/d/4.9.0/std/array.d:1448: note: std.array.split(R1, R2)(R1 r, R2 delim) if (isForwardRange!R1 && isForwardRange!R2 && is(typeof(ElementType!R1.init == ElementType!R2.init)))
/mnt/lotsostuff/dev/tools/x86_64-gdcproject-linux-gnu/include/d/4.9.0/std/array.d:1459: note: std.array.split(alias isTerminator, R)(R r) if (isForwardRange!R && is(typeof(unaryFun!isTerminator(r.front))))
FAIL ../../../../../../../../home/dkords/.dub/packages/serial-port-1.1.0/.dub/build/library-debug-linux.posix-x86_64-gdc_2065-0CC4EDDA44197C04B68AE88DAC1E1B69/ serial-port staticLibrary
Error executing command run: gdc failed with exit code 1.
```

**Comment by NCrashed** [comment]
Can't reproduce with:
```
gdc (crosstool-NG crosstool-ng-1.20.0-232-gc746732 - 20150830-2.066.1-dadb5a3784) 5.2.0
DUB version 0.9.22
```

Could you specify version of GDC the error appeared against?

**Comment by truedat101** [comment]
gdc --version
gdc (crosstool-NG hg+unknown-20140610.075849 - 20140615-2.065-a8ad6a6678) 4.9.0
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

DUB version 0.9.22, built on Sep 16 2014

**Comment by NCrashed** [comment]
Apparently, the 5.2.0 version has the problem fixed. Is that really important to fix the bug under 4.9.0?

**Comment by truedat101** [comment]
No, I'd say that it makes sense to track the current compiler.
